### PR TITLE
[usage] Implement markWorkspaceInstanceSessionAsBilled

### LINKED ISSUE #11884: Implement method `setBilledSession` (for Chargebee) in the Billing Service
We should have method markWorkspaceInstanceSessionAsBilled for Chargebee in the BillingService.

- [x] https://github.com/gitpod-io/gitpod/pull/11909
- [ ] setBilledSession implementation
- [x] #11988

### PR DESCRIPTION
**Pull request by laushinka**
## Description
Implements setBilledSession and calls it during workspace start.

## Related Issue(s)
Depends on #11999
Fixes #11884

## How to test
1. Do NOT create a "Gitpod" team to ensure that you are still under the Chargebee billing mode.
2. Run workspaces either as personal or in a team's project.
3. Check in the DB (I can also check it) that the session is added into `d_b_billed_session`. 

## Release Notes

```release-note
NONE
```

## Werft options:

- [x] /werft with-preview
- [x]  /werft with-payment

### CODE REVIEW

**Comment by roboquat** [comment]
@laushinka: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/server/ee/src/billing/billing-mode.ts
Comment (AlexTugarev) [inline components/server/ee/src/billing/billing-mode.ts:23]: nit: I don't think this is needed as `AttributionId = UserAttributionId | TeamAttributionId` already.

Comment (laushinka) [reply components/server/ee/src/billing/billing-mode.ts:23]: @AlexTugarev Thanks for the catch! This was part of a debugging that accidentally made its way into the commit 🙈 Good thing this is still a draft

File: components/server/src/workspace/workspace-starter.ts
Comment (AlexTugarev) [inline components/server/src/workspace/workspace-starter.ts:849]: @laushinka, I wonder if that call is placed right in `newInstance`, which just pumps up the instance object, but does not do any side-effects otherwise. 

maybe that should be part of `actuallyStartWorkspace`, preferably after the workspace creation request (to wsman) was already done, so it's not gonna be delayed further.

Comment (laushinka) [reply components/server/src/workspace/workspace-starter.ts:849]: Hmm, I guess it makes sense because this call isn't supposed to block or exit the ongoing process.

Comment (laushinka) [reply components/server/src/workspace/workspace-starter.ts]: @AlexTugarev Change done - moved to workspace creation request. Tested and still works fine 👍🏼

File: components/usage/pkg/db/billed_session.go
Comment (andrew-farries) [inline components/usage/pkg/db/billed_session.go:39]: Can inline this to just `return database.Create(&billedSession).Error`

File: components/usage/pkg/db/billed_session.go
Comment (andrew-farries) [inline components/usage/pkg/db/billed_session.go:47]:
```suggestion
		return nil, fmt.Errorf("not found for instance ID: %w", db.Error)
```
error strings shouldn't be capitalised in Go. Also could be improved to say *what* wasn't found.

File: components/usage/pkg/db/billed_session.go
Comment (andrew-farries) [inline components/usage/pkg/db/billed_session.go:42]: This function isn't used anywhere. Should it be part of this PR?

Comment (laushinka) [reply components/usage/pkg/db/billed_session.go:42]: True. I wrote this placeholder method due to [this comment](https://gitpod.slack.com/archives/C02EN94AEPL/p1659600593291589?thread_ts=1659591950.914839&cid=C02EN94AEPL), but you're right it's not necessary. Happy to remove it.

File: components/usage/pkg/apiv1/billing.go
Comment (laushinka) [reply components/usage/pkg/apiv1/billing.go:40]: Oops, thanks!

**Review by andrew-farries — APPROVED** [review]

**Comment by sagor999** [comment]
/hold
to unblock merge queue

**Pull request merged**
